io/pci/dlpar.py

```diff
@@ -88,6 +88,12 @@ def setUp(self):
         if not self.server:
             self.cancel("Managed System not got")
         self.lpar_2 = self.params.get("lpar_2", '*', default=None)
+        cmd = 'lshwres -r io -m %s --rsubtype slot --filter lpar_names=%s ' \
+              '-F drc_index,lpar_id,drc_name,bus_id' % (self.server,
+                                                        self.lpar_2)
+        output = self.run_command(cmd)
+        for line in output:
+            self.lpar2_id = line.split(',')[1]
```

Comment:
`if self.lpar_2:`

Reply:
@smruti77 you need to find lpar2_id only if it is given.

Reply:
ok

Reply:
@narasimhan-v ,but self.lpar2 default is None,if self.lpar_2: this always return true.

Reply:
```
>>> a = '1'
>>> b = None
>>> bool(a)
True
>>> bool(b)
False
>>> 
```